src/api/firebase.js

```diff
@@ -133,3 +121,61 @@ export async function matchToken(listId) {
 	const jointListTokenSnapshot = await getDocs(jointListTokenQuery);
 	return jointListTokenSnapshot;
 }
+
+export function comparePurchaseUrgency(items) {
+	const sortedItems = [];
+
+	items.forEach((item) => {
+		const differenceInDays = item.dateLastPurchased
+			? getDaysBetweenDates(item.dateLastPurchased, item.dateNextPurchased)
+			: getDaysBetweenDates(item.dateCreated, item.dateNextPurchased);
+
+		if (differenceInDays >= 60) {
+			item.isInactive = true;
+			item.urgency = 'Inactive';
+		} else {
+			const differenceTillNextPurchase = getDaysBetweenDates(
+				item.dateNextPurchased,
+			);
+
+			item.isInactive = false;
+			item.days = differenceTillNextPurchase;
+
+			switch (true) {
+				case differenceTillNextPurchase < 0:
+					item.urgency = 'Overdue';
+					break;
+				case differenceTillNextPurchase <= 7:
+					item.urgency = 'Soon';
+					break;
+				case differenceTillNextPurchase > 7 && differenceTillNextPurchase < 30:
+					item.urgency = 'Kind of soon';
+					break;
+				case differenceTillNextPurchase >= 30:
+					item.urgency = 'Not soon';
+					break;
+				default:
+					item.urgency = 'Inactive';
+					break;
+			}
+		}
+
+		if (item.isChecked) {
+			item.urgency = 'Purchased';
+		}
+	});
+
+	items.sort(
```

Comment:
For sorting, perhaps consider more descriptive names than a, b.

Reply:
Also, `isInactive` is not very descriptive. Why `isInactive` vs `isActive`, and what does `isInactive` mean?